### DISCUSSION TITLE
How to resolve local plugin or rule using old rc config file format?

### DISCUSSION THREAD

**Discussion by pawelgur** [eslint/eslint#18365] [Q&A] [answered]
Eslint v8.50, Windows,  using `eslintrc.js` config format

I am following [custom rule tutorial](https://eslint.org/docs/v8.x/extend/custom-rule-tutorial) to create custom rule (which requires a plugin from what I understand), but I'm not able import my local plugin. 

Tried following:
- provide object as per tutorial - object not supported, string required
- try providing path to my plugin - getting `Plugins array cannot includes file paths` error
- create `eslint-plugin-example` dir with index.js in root of the project (next to eslintrc.js) and provide "eslint-plugin-example" in `plugins` config - config not found

The only way I managed to make it work is to do `npm i ./eslint-plugin-example --save-dev` which created symlink in node_modules which looks way too complicated for such simple thing as including local rule/plugin. 

Is there any other way to achieve it? (without migrating to flat config)

**Comment by mdjermanovic** [comment, marked as answer]
> The only way I managed to make it work is to do `npm i ./eslint-plugin-example --save-dev` which created symlink in node_modules which looks way too complicated for such simple thing as including local rule/plugin.

We're doing the same in the eslint project:

https://github.com/eslint/eslint/blob/0f5df509a4bc00cff2c62b90fab184bdf0231322/package.json#L125

It indeed seems too complicated, but it's probably the best way to include local rules when using the `eslintrc` config format. The new flat config format is designed to make including local rules much easier.
